### PR TITLE
Add API client

### PR DESCRIPTION
**Pull request by corvis**

### CODE REVIEW

File: src/openepd/api/sync_client.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
+from requests.auth import AuthBase
+
+from openepd.api.base_sync_client import SyncHttpClient, TokenAuth
+
+__all__ = ("OpenEpdApiClientSync",)
+
```
Comment (alexlitvinenko) [inline src/openepd/api/sync_client.py:25]: reformat `__all__`

File: src/openepd/model/epd.py [hunk 2]
```diff
@@ -127,12 +127,12 @@ class Epd(WithAttachmentsMixin, WithAltIdsMixin, BaseOpenEpdSchema):
...
     )
-    date_of_issue: datetime.date | None = pyd.Field(
-        example=datetime.date(day=11, month=9, year=2019),
+    date_of_issue: datetime.datetime | None = pyd.Field(
+        example=datetime.datetime(day=11, month=9, year=2019, tzinfo=datetime.timezone.utc),
         description="Date the EPD was issued. This should be the first day on which the EPD is valid.",
     )
-    valid_until: datetime.date | None = pyd.Field(
-        example=datetime.date(day=11, month=9, year=2028),
+    valid_until: datetime.datetime | None = pyd.Field(
```
Comment (alexlitvinenko) [inline src/openepd/model/epd.py:134]: Describe datetime format.

File: src/openepd/model/pcr.py [hunk 3]
```diff
@@ -70,13 +80,28 @@ class Pcr(WithAttachmentsMixin, WithAltIdsMixin, BaseOpenEpdSchema):
...
+        "more than 5 years in the past must have status Sunset. Compliant systems should automatically "
+        "update these fields within 24 hours.",
+    )
     product_classes: dict[str, str | list[str]] = pyd.Field(
         description="List of classifications, including Masterformat and UNSPC", default_factory=dict
     )
+    applicable_in: list[Annotated[str, pyd.Field(min_length=2, max_length=2)]] | None = pyd.Field(
+        max_items=100,
+        default=None,
+        description="Jurisdiction(s) in which EPD is applicable. An empty array, or absent properties, "
```
Comment (alexlitvinenko) [inline src/openepd/model/pcr.py:101]: Describe the format of a list item.

**Review by alexlitvinenko — CHANGES_REQUESTED** [review]

**Review by irynadosiak-c-change-lab — APPROVED** [review]

File: src/openepd/api/epd/dto.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+class StatisticsDto(BaseOpenEpdApiModel):
+    """
+    DTO describes statistics of EPD.
+
+    Statistics contains aggregated parameters such as percentiles distributions of GWP and other parameters.
+    """
+
+    # percentiles
+    pct10_gwp: float = pyd.Field("10th percentile GWP for this statistics")
```
[the remark this answers is not in the document]
Comment (alexche-cchange) [reply src/openepd/api/epd/dto.py:40]: @alexlitvinenko I remember us recently having very same discussion about using floats for simple_baseline, and the conclusion was that everyone understands this is in kgCO2e, and we can safely use floats. 

Let's maybe just make it in comment - i.e. add `pyd.Field("10th percentile GWP for this statistics, in kg CO2 equivalent")`. 

Do you think that would be legit?

[the remark this answers is not in the document]
Comment (alexlitvinenko) [reply src/openepd/api/epd/dto.py:40]: Yes, that is what I meant. Just to describe in description.

File: src/openepd/api/common.py [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+            start_from_page = 1
+        self.goto_page(start_from_page)
+        while True:
+            items = self.goto_page(self.current_page)
+            for x in items:
+                yield x
+            if not self.has_next_page():
+                return  # no more pages
+            else:
+                self.goto_page(self.current_page + 1)
```
[the remark this answers is not in the document]
Comment (corvis) [reply src/openepd/api/common.py:216]: Goto page does nothing is target page == current page, just returns result as list

File: src/openepd/api/epd/dto.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (corvis) [reply src/openepd/api/epd/dto.py]: Updated description

File: src/openepd/api/sync_client.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        :param kwargs: additional arguments to pass to the HTTP client. See SyncHttpClient constructor for details.
+        """
+        super().__init__()
+        auth: AuthBase | None = TokenAuth(auth_token) if auth_token is not None else None
+        self._http_client = SyncHttpClient(base_url, auth=auth, **kwargs)
+        self.__epd_api: EpdApi | None = None
+        self.__category_api: CategoryApi | None = None
+
+    @property
+    def epd(self) -> EpdApi:
```
[the remark this answers is not in the document]
Comment (corvis) [reply src/openepd/api/sync_client.py]: done

File: src/openepd/model/epd.py [hunk 7]
```diff
@@ -127,12 +127,12 @@ class Epd(WithAttachmentsMixin, WithAltIdsMixin, BaseOpenEpdSchema):
     third_party_verifier_email: pyd.EmailStr | None = pyd.Field(
         description="Email address of the third party verifier", example="john.doe@example.com", default=None
     )
-    date_of_issue: datetime.date | None = pyd.Field(
-        example=datetime.date(day=11, month=9, year=2019),
+    date_of_issue: datetime.datetime | None = pyd.Field(
+        example=datetime.datetime(day=11, month=9, year=2019, tzinfo=datetime.timezone.utc),
         description="Date the EPD was issued. This should be the first day on which the EPD is valid.",
```
[the remark this answers is not in the document]
Comment (corvis) [reply src/openepd/model/epd.py:132]: It should be clear from the example (will be generated automatically during swager generation)

File: src/openepd/model/pcr.py (hunk 3 above)
Comment (corvis) [reply src/openepd/model/pcr.py:101]: done

File: src/openepd/model/epd.py (hunk 2 above)
Comment (corvis) [reply src/openepd/model/epd.py:134]: see above

File: src/openepd/api/base_sync_client.py [hunk 8]
```diff
@@ -0,0 +1,533 @@
...
+
+class TokenAuth(AuthBase):
+    """Attach OpenEPD Token Authentication to the given Request object."""
+
+    def __init__(self, token: str) -> None:
+        super().__init__()
+        self.token = token
+
+    def __eq__(self, other):
+        return self.token == other.token
```
[the remark this answers is not in the document]
Comment (corvis) [reply src/openepd/api/base_sync_client.py:141]: Done

File: src/openepd/api/errors.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    """Server error."""
+
+    pass
+
+
+class ValidationError(ApiError):
+    """Validation error."""
+
+    def __init__(
+        self, http_status: int, error_summary: str, validation_errors: dict[str, list[str]] | None, response, error_code
```
Comment (alexlitvinenko) [inline src/openepd/api/errors.py:52]: type hint response and error_code

File: src/openepd/api/pcr/sync_api.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+from openepd.api.pcr.dto import PcrRef
+from openepd.model.pcr import Pcr
+
+
+class PcrApi(BaseApiMethodGroup):
+    """API methods for EPDs."""
+
+    def get_by_openxpd_uuid(self, uuid: str) -> Pcr:
+        """
+        Get PCR by OpenEPD UUID.
```
Comment (alexlitvinenko) [inline src/openepd/api/pcr/sync_api.py:30]:
```suggestion
        Get PCR by Open xPD UUID.
```

File: src/openepd/api/pcr/sync_api.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+        :param uuid: OpenEPD UUID
```
Comment (alexlitvinenko) [inline src/openepd/api/pcr/sync_api.py:32]:
```suggestion
        :param uuid: Open xPD UUID
```

File: src/openepd/api/pcr/sync_api.py [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        :return: EPD
```
Comment (alexlitvinenko) [inline src/openepd/api/pcr/sync_api.py:33]:
```suggestion
        :return: PCR
```

**Review by alexlitvinenko — CHANGES_REQUESTED** [review]

File: src/openepd/model/pcr.py [hunk 13]
```diff
@@ -70,13 +80,29 @@ class Pcr(WithAttachmentsMixin, WithAltIdsMixin, BaseOpenEpdSchema):
...
+        default=None,
+        description="The current release status of this PCR. "
+        "A PCR with valid_until in the past must have status Expired or Sunset; a PCR with valid_until "
+        "more than 5 years in the past must have status Sunset. Compliant systems should automatically "
+        "update these fields within 24 hours.",
+    )
     product_classes: dict[str, str | list[str]] = pyd.Field(
         description="List of classifications, including Masterformat and UNSPC", default_factory=dict
     )
+    applicable_in: list[Annotated[str, pyd.Field(min_length=2, max_length=2)]] | None = pyd.Field(
```
Comment (alexlitvinenko) [inline src/openepd/model/pcr.py:98]: EPD.applicable_in and PCR.applicable_in are accepting the same value:
The most recent description:
image
https://github.com/cchangelabs/openepd-openapi/blob/dc720e754611b39961f0f1e2af50ec42b37d650d/content/schemas/models/EPD/Base.yaml#L143

File: src/openepd/api/dto/common.py [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+
+from openepd.api.dto.base import BaseMetaDto, BaseOpenEpdApiModel, MetaExtensionBase
+from openepd.api.dto.meta import PerformanceMetaMixin
+from openepd.model.base import AnySerializable, BaseOpenEpdSchema
+
+DEFAULT_PAGE_SIZE: Final[int] = 100
+MAX_PAGE_SIZE: Final[int] = 250
+
+
+class AuditableDto(BaseOpenEpdApiModel, metaclass=abc.ABCMeta):
```
Comment (alexche-cchange) [inline src/openepd/api/dto/common.py:35]: Do you think we'll be able to replace the local version of same with this? E.g. the serverside.

**Review by alexche-cchange — APPROVED** [review]
looks good

**Pull request merged**